straxen/get_corrections.py

```diff
@@ -34,152 +36,75 @@ def correction_options_wrapped(run_id, conf, *arg):
-                             '("CMT_model", ("to_pe_model", "v1"). Instead got:'
-                             f'{model_conf}')
-        # is this the best way to do this?
-        is_nt = n_pmts == straxen.n_tpc_pmts or n_pmts == straxen.n_nveto_pmts or n_pmts == straxen.n_mveto_pmts
-
-        corrections = straxen.CorrectionsManagementServices(is_nt=is_nt)
-        to_pe = corrections.get_corrections_config(run_id, model_conf)
-
-        return to_pe
+def get_correction_from_cmt(run_id, conf):
+    """
+    Get correction from CMT general format is
+    conf = ('correction_name', 'version', True)
+    where True means looking at nT runs, e.g. 
+    get_correction_from_cmt(run_id, conf[:2])
+    special cases:
+    version can be replaced by consant int, float or array
+    when user specify value(s)
+    :param run_id: run id from runDB
+    :param conf: configuration 
+    :return: correction value(s)
+    """
+ 
+    if isinstance(conf, str) and conf.startswith('https://raw'):
+        # Legacy support for pax files
+        return conf
 
-    elif model_type == 'to_pe_per_run':
-        warn("to_pe_per_run will be replaced by CorrectionsManagementSevices",
-             DeprecationWarning, 2)
-        # Load a npy file specifying a run_id -> to_pe array
-        to_pe_file = model_conf
-        x = straxen.get_resource(to_pe_file, fmt='npy')
-        run_index = np.where(x['run_id'] == int(run_id))[0]
-        if not len(run_index):
-            # Gains not known: using placeholders
-            run_index = [-1]
-        to_pe = x[run_index[0]]['to_pe']
+    elif isinstance(conf, tuple) and len(conf) == 2:
+        model_conf, cte_value = conf[:2]
 
-    elif model_type == 'to_pe_constant':
+        # special case constant to_pe values 
         if model_conf in FIXED_TO_PE:
-            return FIXED_TO_PE[model_conf]
-
-        try:
-            # Uniform gain, specified as a to_pe factor
-            to_pe = np.ones(n_pmts, dtype=np.float32) * model_conf
-        except np.core._exceptions.UFuncTypeError as e:
-            raise (str(e) +
-                   f"\nTried multiplying by {model_conf}. Insert a number instead.")
-    else:
-        raise NotImplementedError(f"Gain model type {model_type} not implemented")
+            correction = FIXED_TO_PE[model_conf]
+            return correction
 
-    if len(to_pe) != n_pmts:
-        raise ValueError(
-            f"Gain model {gain_model} resulted in a to_pe "
-            f"of length {len(to_pe)}, but n_pmts is {n_pmts}!")
-    return to_pe
+        # special case constant single value
+        elif 'constant' in model_conf:
+            if not isinstance(cte_value, (float, int)):
+                raise ValueError(f"User specify a model type {model_conf} "
+                                 "and should provide a number. Got: "
+                                 f"{type(cte_value)}")
+            correction = cte_value
+            return correction
```

Comment:
No, comment. I just need to have some thing I can link in https://github.com/XENONnT/mc_chain/pull/14.

Reply:
I apologize, there was not any other way of linking to the code in this PR directly. I was discussion with Diego something about https://github.com/XENONnT/mc_chain/pull/14.

Reply:
no worries